src/store/flow/resource.ts

```diff
@@ -28,48 +27,51 @@ import {IFlowsState} from '@/store/flow/index'
 import {IRootState} from '@/store'
 
 export const getters: GetterTree<IFlowsState, IRootState> = {
-  resourcesByUuid: ({resources}) => keyBy(resources, 'uuid'),
+  resourcesByUuidOnActiveFlow: (_state, getters) => keyBy(getters.activeFlow.resources, 'uuid'),
 
+  //TODO: try to redefine this using getters.activeFlow.resources only
   resourceUuidsOnActiveFlow: (state, getters) => filter(map(getters.activeFlow.blocks, (block) => block.config.prompt)),
-
-  resourcesOnActiveFlow: (state, getters): IResource[] => filter(
-    state.resources,
-    (resource) => includes(getters.resourceUuidsOnActiveFlow, resource.uuid)
-  ),
 }
 
-export const mutations: MutationTree<IFlowsState> = {
-  resource_add({resources}, {resource}: { resource: IResource }) {
```

Comment:
Moved these into actions, because we have access to `getters` there

Reply:
See my comment about this.

Reply:
OK 👁️